covalent/_serialize/result.py

```diff
@@ -0,0 +1,205 @@
+        "_status": Status(meta.status),
+        "_start_time": meta.start_time,
+        "_end_time": meta.end_time,
+    }
+
+
+def _serialize_result_assets(res: Result, storage_path: str) -> ResultAssets:
+    # NOTE: We can avoid pickling here since the UI actually consumes only the string representation
+
+    error_asset = save_asset(
+        res._error, ASSET_TYPES["error"], storage_path, ASSET_FILENAME_MAP["error"]
+    )
+    result_asset = save_asset(
+        res._result,
+        ASSET_TYPES["result"],
+        storage_path,
+        ASSET_FILENAME_MAP["result"],
+    )
+    return ResultAssets(result=result_asset, error=error_asset)
+
+
+def _deserialize_result_assets(assets: ResultAssets) -> dict:
+    error = load_asset(assets.error, ASSET_TYPES["error"])
+    result = load_asset(assets.result, ASSET_TYPES["result"])
+    return {"_result": result, "_error": error}
+
+
+def serialize_result(res: Result, storage_path: str) -> ResultSchema:
+    meta = _serialize_result_metadata(res)
+    assets = _serialize_result_assets(res, storage_path)
+    lat = serialize_lattice(res.lattice, storage_path)
+    return ResultSchema(metadata=meta, assets=assets, lattice=lat)
+
+
+def deserialize_result(res: ResultSchema) -> Result:
+    dispatch_id = res.metadata.dispatch_id
+    lat = deserialize_lattice(res.lattice)
+    result_object = Result(lat, dispatch_id)
+    attrs = _deserialize_result_metadata(res.metadata)
+    assets = _deserialize_result_assets(res.assets)
+
+    attrs.update(assets)
+    result_object.__dict__.update(attrs)
+    return result_object
+
+
+# Functions to preprocess manifest for submission
+
+
+def strip_local_uris(res: ResultSchema) -> ResultSchema:
+    # Create a copy with the local uris removed for submission
+    manifest = res.copy(deep=True).dict()
+
+    # Strip workflow asset uris:
+    dispatch_assets = manifest["assets"]
+    for _, asset in dispatch_assets.items():
+        asset["uri"] = ""
+
+    lattice = manifest["lattice"]
+    lattice_assets = lattice["assets"]
+    for _, asset in lattice_assets.items():
+        asset["uri"] = ""
+
+    # Node assets
+    tg = lattice["transport_graph"]
+
+    nodes = tg["nodes"]
+    for node in nodes:
+        node_assets = node["assets"]
+        for _, asset in node_assets.items():
+            asset["uri"] = ""
+
+    return ResultSchema.parse_obj(manifest)
+
+
+# Functions to postprocess response from dispatcher
+
+
+def merge_response_manifest(manifest: ResultSchema, response: ResultSchema) -> ResultSchema:
+    """Merge the dispatcher's response with the submitted manifest.
+
+    Args:
+        manifest: The manifest submitted to the `/register` endpoint.
+        response: The manifest returned from `/register`.
+    Returns:
+        A combined manifest with asset `remote_uri`s populated.
+    """
+
+    manifest.metadata.dispatch_id = response.metadata.dispatch_id
+    manifest.metadata.root_dispatch_id = response.metadata.root_dispatch_id
```

Comment:
```suggestion
    Returns:
        A combined manifest with asset `remote_uri`s populated.
        
    """
    manifest.metadata.dispatch_id = response.metadata.dispatch_id
    manifest.metadata.root_dispatch_id = response.metadata.root_dispatch_id
```

(Non-blocking trivial comment) I think the doc strings look better this way. Also the style used in the other link I sent you. In either case as long as there's an empty line before OR after the quotes it's kosher.